### PR TITLE
feat(x/ecocredit)!: remove basket multiplier and simplify picking

### PR DESCRIPTION
**Pull request by aaronc**
## Description

Closes: #641 #640 

This PR:
* removes differential multipliers from baskets
* enforces that baskets can only contain one credit type (ex. "carbon" or "biodiversity" but not both)
* specifies that allow picking either allows everyone to pick or nobody
* adds a fee for picking operations
* fixes some issues with disable_auto_retire semantics

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by technicallyty — APPROVED** [review]

File: proto/regen/ecocredit/v1alpha2/types.proto [hunk 1]
```diff
@@ -230,36 +230,35 @@ message Filter {
     And and = 1;
     Or or = 2;
 
-    string credit_type_name = 3;
-    string class_id = 4;
+    string class_id = 3;
```
Comment (ryanchristo) [inline proto/regen/ecocredit/v1alpha2/types.proto:233]:
```suggestion
    // class_id filters against credits from this batch.
    string class_id = 3;
```

File: proto/regen/ecocredit/v1alpha2/types.proto [hunk 2]
```diff
@@ -230,36 +230,35 @@ message Filter {
...
 
     // project_id filters against credits from this batch.
-    string project_id = 5;
+    string project_id = 4;
 
     // batch_id filters against credits from this batch.
```
Comment (ryanchristo) [inline proto/regen/ecocredit/v1alpha2/types.proto:238]:
```suggestion
    // batch_denom filters against credits from this batch.
```

File: proto/regen/ecocredit/v1alpha2/types.proto [hunk 3]
```diff
@@ -230,36 +230,35 @@ message Filter {
...
     // country, sub-national-code, or postal code. If just country is given,
     // for instance "US" then any credits in the "US" will be matched even
     // their project location is more specific, ex. "US-NY 12345". If
     // a country, sub-national-code and postal code are all provided then
     // only projects in that postal code will match.
-    string project_location = 10;
+    string project_location = 9;
 
-    DateRange date_range = 11;
+    DateRange date_range = 10;
```
Comment (ryanchristo) [inline proto/regen/ecocredit/v1alpha2/types.proto:258]:
```suggestion
    // date_range is the date range in which credits from this credit
    // batch were quantified and verified.
    DateRange date_range = 10;
```

**Review by ryanchristo — APPROVED** [review]
Looks good to me. A couple suggestions adding/fixing comments but it looks like they might also be covered within #628. Pre-approving.

**Comment by aaronc** [comment]
> @aaronc If you're busy, maybe @ryanchristo or @technicallyty can make the required changes,, and get this PR merged?

Thanks @amaurym, but it actually turns out the whole basket API is being simplified based on changing requirement, so closing this for now

**Pull request closed without merging**